src/nav/nav.ts

```diff
@@ -196,6 +214,68 @@ export class NgbNav implements AfterContentInit {
     }
   }
 
+  onKeyDown(event: KeyboardEvent) {
+    if (this.roles !== 'tablist') {
+      return;
+    }
+    // tslint:disable-next-line: deprecation
+    const key = event.which;
+    const tabs: any[] = this.items.filter(x => !x.disabled);
+
+    const navLinkElements = tabs.map(
+        x => x.isCommentNode() ? x.elementRef.nativeElement.nextSibling :
+                                 x.elementRef.nativeElement.querySelector('.nav-link'));
+
+    let position = -1;
+
+    navLinkElements.forEach((element, index) => {
+      if (element === this._document.activeElement) {
+        position = index;
+      }
+    });
+
+    if (navLinkElements.length) {
```

Comment:
Let's maybe factor this out as `const {length} = enabledLinks;` as I suggested in another comment